recovery_services_vault while setting up immutability with existing recovery service vault (LRS and GRS)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.0

### AzureRM Provider Version

3.93.0

### Affected Resource(s)/Data Source(s)

azurerm_recovery_services_vault

### Terraform Configuration Files

```hcl
## the code for existing rsv

resource "azurerm_recovery_services_vault" "example" {
  location            = "westeurope"
  resource_group_name = "example"
  name                = "test-lrs-rv"
  sku                 = "Standard"
  storage_mode_type   = "LocallyRedundant"
  tags                = local.tags
  
 + immutability        = "Locked"
}
```


### Debug Output/Panic Output

```shell
Error: Plugin did not respond
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may

github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices.resourceRecoveryServicesVaultUpdate(0x0?, {0x7a19160?, 0xc000b62480})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices/recovery_services_vault_resource.go:578 +0x1b95
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x922bad0?, {0x922bad0?, 0xc0031e7620?}, 0xd?, {0x7a19160?, 0xc000b62480?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:800 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001845420, {0x922bad0, 0xc0031e7620}, 0xc001771930, 0xc00155d500, {0x7a19160, 0xc000b62480})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0009cf1a0, {0x922bad0?, 0xc0031e7530?}, 0xc0031d3950)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001cd65a0, {0x922bad0?, 0xc0031e6d20?}, 0xc000783960)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x85fb5e0?, 0xc001cd65a0}, {0x922bad0, 0xc0031e6d20}, 0xc0007838f0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002121e0, {0x9252ba0, 0xc001d331e0}, 0xc0031f0360, 0xc001d72990, 0xec16eb8, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc0002121e0, {0x9252ba0, 0xc001d331e0}, 0xc0031f0360, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 14
	google.golang.org/grpc@v1.58.3/server.go:997 +0x145

Error: The terraform-provider-azurerm_v3.93.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

Error: Process completed with exit code 1.
```

### Expected Behaviour

It should enable immutable setting on RSV

### Actual Behaviour

plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may

### Steps to Reproduce

terraform apply

### Important Factoids

westeurope

### References

_No response_

--- a/internal/services/recoveryservices/recovery_services_vault_resource.go
+++ b/internal/services/recoveryservices/recovery_services_vault_resource.go
@@ -594,9 +594,14 @@ func resourceRecoveryServicesVaultUpdate(d *pluginsdk.ResourceData, meta interfa
 		// Hence we firstly make it `Unlocked`, and once created, we will update it to `Locked`.
 		// Note: The `immutability` could be transitioned only in the limited directions.
 		// Locked <- Unlocked <-> Disabled
-		currentImmutability := model.Properties.SecuritySettings.ImmutabilitySettings.State
+
+		// When the service returns `null`, it equals `disabled`
+		currentImmutability := pointer.To(vaults.ImmutabilityStateDisabled)
+		if model.Properties != nil && model.Properties.SecuritySettings != nil && model.Properties.SecuritySettings.ImmutabilitySettings != nil && model.Properties.SecuritySettings.ImmutabilitySettings.State != nil {
+			currentImmutability = model.Properties.SecuritySettings.ImmutabilitySettings.State
+		}
 		immutability := d.Get("immutability")
-		if currentImmutability != nil && string(*currentImmutability) == string(vaults.ImmutabilityStateDisabled) && immutability == string(vaults.ImmutabilityStateLocked) {
+		if string(*currentImmutability) == string(vaults.ImmutabilityStateDisabled) && immutability == string(vaults.ImmutabilityStateLocked) {
 			additionalUpdatePatch.Properties.SecuritySettings = expandRecoveryServicesVaultSecuritySettings(immutability)
 			requireAdditionalUpdate = true
 			immutability = string(vaults.ImmutabilityStateUnlocked)

--- a/internal/services/recoveryservices/recovery_services_vault_resource_test.go
+++ b/internal/services/recoveryservices/recovery_services_vault_resource_test.go
@@ -231,6 +231,13 @@ func TestAccRecoveryServicesVault_immutability(t *testing.T) {
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
 			Config: r.basicWithImmutability(data, "Unlocked"),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),

